### PR TITLE
Read partitioned parquet files

### LINKED ISSUE #624: Unable to read entire parquet directory
<!--
General questions should be asked on the mailing list modin-dev@googlegroups.com.

Before submitting an issue, please fill out the following form.
-->

### System information
- Ubuntu 19.04
- pipenv install modin:
- **Modin version**: 0.5.0
- **Python version**: 3.7.3
- **Exact command to reproduce**: pd.read_parquet('parquet-out/foo/')

<!--
You can obtain the Modin version with

python -c "import modin; print(modin.__version__)"
-->

### Describe the problem
<!-- Describe the problem clearly here. -->
It seems that read_parquet cannot read an entire directory of parquet files i.e. partitioned parquet directory. 

### Source code / logs
<!-- Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem. -->

~/.local/share/virtualenvs/foo-cFmMv4Vf/lib/python3.7/site-packages/modin/pandas/io.py in read_parquet(path, engine, columns, **kwargs)                                                                         
     27     return DataFrame(                                                                                                                                                                                      
     28         query_compiler=BaseFactory.read_parquet(                                                                                                                                                           
---> 29             path=path, columns=columns, engine=engine, **kwargs                                                                                                                                            
     30         )                                                                                                                                                                                                  
     31     )                                                                                                                                                                                                      
                                                                                                                                                                                                                   
~/.local/share/virtualenvs/foo-cFmMv4Vf/lib/python3.7/site-packages/modin/data_management/factories.py in read_parquet(cls, **kwargs)                                                                           
     49     @classmethod                                                                                                                                                                                           
     50     def read_parquet(cls, **kwargs):                                                                                                                                                                       
---> 51         return cls._determine_engine()._read_parquet(**kwargs)                                                                                                                                             
     52                                                                                                                                                                                                            
     53     @classmethod                                                                                                                                                                                           
                                                                                                                                                                                                                   
~/.local/share/virtualenvs/foo-cFmMv4Vf/lib/python3.7/site-packages/modin/data_management/factories.py in _read_parquet(cls, **kwargs)                                                                          
     53     @classmethod                                                                                                                                                                                           
     54     def _read_parquet(cls, **kwargs):                                                                                                                                                                      
---> 55         return cls.io_cls.read_parquet(**kwargs)                                                                                                                                                           
     56                                                                                                                                                                                                            
     57     @classmethod                                                                                                                                                                                           
                                                                                                                                                                                                                   
~/.local/share/virtualenvs/foo-cFmMv4Vf/lib/python3.7/site-packages/modin/engines/ray/generic/io.py in read_parquet(cls, path, engine, columns, **kwargs)                                                       
    149                                                                                                                                                                                                            
    150         if not columns:                                                                                                                                                                                   
--> 151             pf = ParquetFile(path, memory_map=False)                                                                                                                                                      
    152             columns = [                                                                                                                                                                                   
    153                 name                                                                                                                                              
                                                                                                                                                                                                                  
~/.local/share/virtualenvs/foo-cFmMv4Vf/lib/python3.7/site-packages/ray/pyarrow_files/pyarrow/parquet.py in __init__(self, source, metadata, common_metadata, memory_map)
    127                  memory_map=True):                                                                                                                                                                        
    128         self.reader = ParquetReader()                                                                                                                                                                     
--> 129         self.reader.open(source, use_memory_map=memory_map, metadata=metadata)                                                                                                                            
    130         self.common_metadata = common_metadata                                                                                                                                                             
    131         self._nested_paths_by_prefix = self._build_nested_paths()

~/.local/share/virtualenvs/foo-cFmMv4Vf/lib/python3.7/site-packages/ray/pyarrow_files/pyarrow/_parquet.pyx in pyarrow._parquet.ParquetReader.open()

~/.local/share/virtualenvs/foo-cFmMv4Vf/lib/python3.7/site-packages/ray/pyarrow_files/pyarrow/error.pxi in pyarrow.lib.check_status()
ArrowIOError: Arrow error: IOError: Error reading bytes from file: Is a directory

### PR DESCRIPTION
**Pull request by williamma12**
## What do these changes do?

Adds special case for partitioned parquet files

## Related issue number

Fixes #624 and #633 

- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] tests added and passing

### CODE REVIEW

**Comment by ddutt** [comment]
Hi @williamma12,

When I applied your fixes to my 0.5.0 tree, I get a different error (I've turned off memory-map):

RayTaskError: ray_worker (pid=31117, host=ddutt-yoga)
  File "/home/ddutt/.local/share/virtualenvs/a-cFmMv4Vf/lib/python3.7/site-packages/modin/engines/ray/pandas_on_ray/io.py", line 55, in _read_parquet_columns                                                
    df = pq.read_pandas(path, columns=columns, **kwargs).to_pandas()
  File "/home/ddutt/.local/share/virtualenvs/a-cFmMv4Vf/lib/python3.7/site-packages/ray/pyarrow_files/pyarrow/parquet.py", line 1144, in read_pandas                                                         
    use_pandas_metadata=True)
  File "/home/ddutt/.local/share/virtualenvs/a-cFmMv4Vf/lib/python3.7/site-packages/ray/pyarrow_files/pyarrow/parquet.py", line 1123, in read_table                                                          
    use_pandas_metadata=use_pandas_metadata)
  File "/home/ddutt/.local/share/virtualenvs/a-cFmMv4Vf/lib/python3.7/site-packages/ray/pyarrow_files/pyarrow/filesystem.py", line 181, in read_parquet                                                      
    use_pandas_metadata=use_pandas_metadata)
  File "/home/ddutt/.local/share/virtualenvs/a-cFmMv4Vf/lib/python3.7/site-packages/ray/pyarrow_files/pyarrow/parquet.py", line 985, in read                                                                 
    use_pandas_metadata=use_pandas_metadata)
  File "/home/ddutt/.local/share/virtualenvs/a-cFmMv4Vf/lib/python3.7/site-packages/ray/pyarrow_files/pyarrow/parquet.py", line 535, in read                                                                 
    table = reader.read(**options)
  File "/home/ddutt/.local/share/virtualenvs/a-cFmMv4Vf/lib/python3.7/site-packages/ray/pyarrow_files/pyarrow/parquet.py", line 211, in read                                                                 
    columns, use_pandas_metadata=use_pandas_metadata)
  File "/home/ddutt/.local/share/virtualenvs/a-cFmMv4Vf/lib/python3.7/site-packages/ray/pyarrow_files/pyarrow/parquet.py", line 261, in _get_column_indices                                                  
    for descr in index_columns
  File "/home/ddutt/.local/share/virtualenvs/a-cFmMv4Vf/lib/python3.7/site-packages/ray/pyarrow_files/pyarrow/parquet.py", line 262, in <listcomp>                                                           
    if descr['kind'] == 'serialized']
TypeError: string indices must be integers

File: modin/engines/ray/pandas_on_ray/io.py [hunk 1]
```diff
@@ -52,6 +52,7 @@ def _read_parquet_columns(path, columns, num_splits, kwargs):  # pragma: no cove
     import pyarrow.parquet as pq
 
     df = pq.read_pandas(path, columns=columns, **kwargs).to_pandas()
+    df = df[columns]
```
Comment (devin-petersohn) [inline modin/engines/ray/pandas_on_ray/io.py]: Just delete this. It will be fixed when the resolution for #636 is added.

Comment (williamma12) [reply modin/engines/ray/pandas_on_ray/io.py]: sounds good! This will be blocked on #638

**Comment by williamma12** [comment]
Blocked on #638

**Pull request merged**